### PR TITLE
[KOGITO-2372] - Shared libaries: Git commit and push

### PR DESCRIPTION
**Pull request by Kaitou786**
Added function for creating branch and commit changes.
Push functionality is part of PR #41

SEE: https://issues.redhat.com/browse/KOGITO-2372

Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

### CODE REVIEW

File: vars/githubscm.groovy [hunk 1]
```diff
@@ -74,6 +74,24 @@ def mergeSourceIntoTarget(String repository, String sourceAuthor, String sourceB
...
 }
 
+def createBranch(String branchName){
+    try {
+        sh "git checkout -b ${branchName}"
+    } catch (Exception e) {
+        println "[ERROR] Can't create branch ${branchName} on repo."
+        throw e;
+    }
+ println "[INFO] Created branch '${branchName}' on repo."
```
Comment (Ginxo) [inline vars/githubscm.groovy:84]: Indentation here

File: vars/githubscm.groovy [hunk 2]
```diff
@@ -74,6 +74,24 @@ def mergeSourceIntoTarget(String repository, String sourceAuthor, String sourceB
...
+}
+
+def commitChanges(String userName, String userEmail, String commitMessage){
```
Comment (Ginxo) [inline vars/githubscm.groovy:87]: I would replace this method by

```
def commitChanges(String userName, String userEmail, String commitMessage, String filesToAdd = '--all'){
    sh "git config user.name '${userName}'"
    sh "git config user.email '${userEmail}' "
    sh "git add ${filesToAdd}"
    sh "git commit -m '${commitMessage}' "
}
```
WDYT?

Comment (Kaitou786) [reply vars/githubscm.groovy:87]: Yes, makes more sense.

Comment (Kaitou786) [reply vars/githubscm.groovy]: Updated

**Comment by Ginxo** [comment]
@Kaitou786 please resolve conflicts, thanks

**Pull request merged**